.github/workflows/pytest.yml

```diff
@@ -0,0 +1,28 @@
+name: Run Unit Test via Pytest  
+  
+on: [push]  
+  
+jobs:  
+  build:  
+    runs-on: ubuntu-latest  
+    strategy:  
+      matrix:  
+        python-version: ["3.10"]  
+  
+    steps:  
+      - uses: actions/checkout@v3  
+      - name: Set up Python ${{ matrix.python-version }}  
+        uses: actions/setup-python@v4  
+        with:  
+          python-version: ${{ matrix.python-version }}  
+      - name: Install dependencies  
+        run: |  
+          python -m pip install --upgrade pip
+          pip install testtools
+          pip install requests
+          pip install pytest
+          pip install pytest-coverage
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Test with pytest  
+        run: |  
+          python -m unittest -v tests/test_*.py
```

Comment:
since we are installing pytest, shud we use -m pytest?

Reply:
Running the tests with pytest needs changes in the directory structure, hence continuing with python unittest